crate_universe/src/splicing.rs

```diff
@@ -294,80 +296,83 @@ impl WorkspaceMetadata {
         let crate_indexes = index_urls
             .into_iter()
             .map(|url| {
-                let index = {
-                    // Ensure the correct registry is mapped based on the give Cargo config.
-                    let index_url = if let Some(config) = &cargo_config {
-                        if let Some(source) = config.get_source_from_url(&url) {
-                            if let Some(replace_with) = &source.replace_with {
-                                if let Some(replacement) = config.get_registry_index_url_by_name(replace_with) {
-                                    replacement
-                                } else {
-                                    bail!("Tried to replace registry {} with registry named {} but didn't have metadata about the replacement", url, replace_with);
-                                }
+                // Ensure the correct registry is mapped based on the give Cargo config.
+                let index_url = if let Some(config) = &cargo_config {
+                    if let Some(source) = config.get_source_from_url(&url) {
+                        if let Some(replace_with) = &source.replace_with {
+                            if let Some(replacement) = config.get_registry_index_url_by_name(replace_with) {
+                                replacement
                             } else {
-                                &url
+                                bail!("Tried to replace registry {} with registry named {} but didn't have metadata about the replacement", url, replace_with);
                             }
                         } else {
                             &url
                         }
                     } else {
                         &url
+                    }
+                } else {
+                    &url
+                };
+                let index = if cargo.use_sparse_registries_for_crates_io()? && index_url == "https://github.com/rust-lang/crates.io-index" {
```

Comment:
Perhaps this string could live as a global somewhere and also be used here? https://github.com/bazelbuild/rules_rust/blob/0.18.0/crate_universe/src/splicing/cargo_config.rs#L35-L38

Reply:
Good call! Done!